src/posix/platform/config.cpp

```diff
@@ -78,80 +76,161 @@ otError PowerUpdater::SetRegion(uint16_t aRegionCode)
+    {
+        if ((str = strtok(value, kCommaDelimiter)) == nullptr)
+        {
+            continue;
+        }
 
-    // If the distribution of output power is large, the factory needs to measure the power calibration data
-    // for each device individually, and the power calibration data will be written to the factory config file.
-    // Otherwise, the power calibration data can be pre-configured in the product config file.
-    if (calibrationFile->Get(kKeyCalibratedPower, iterator, value, sizeof(value)) != OT_ERROR_NONE)
+        while ((str = strtok(nullptr, kCommaDelimiter)) != nullptr)
+        {
+            if ((strlen(str) == 2) && (StringToRegionCode(str) == aRegionCode))
+            {
+                ExitNow(error = aDomain.Set(value));
+            }
+        }
+    }
+
+exit:
+    if (error != OT_ERROR_NONE)
     {
-        calibrationFile = &mProductConfigFile;
+        otLogCritPlat("Get domain failed, Error: %s", otThreadErrorToString(error));
     }
 
-    iterator = 0;
-    while (calibrationFile->Get(kKeyCalibratedPower, iterator, value, sizeof(value)) == OT_ERROR_NONE)
+    return error;
+}
+
+otError Config::GetChannelMask(const char *aKey, const Power::Domain &aDomain, uint32_t &aChannelMask)
+{
+    otError       error    = OT_ERROR_NOT_FOUND;
+    int           iterator = 0;
+    char          value[kMaxValueSize];
+    char         *str;
+    Power::Domain domain;
+    uint32_t      channelMask;
+
+    while (mProductConfigFile.Get(aKey, iterator, value, sizeof(value)) == OT_ERROR_NONE)
     {
-        SuccessOrExit(error = calibratedPower.FromString(value));
-        otLogInfoPlat("Update calibrated power: %s\r\n", calibratedPower.ToString().AsCString());
+        if (((str = strtok(value, kCommaDelimiter)) == nullptr) || (aDomain != str))
+        {
+            continue;
+        }
 
-        for (uint8_t ch = calibratedPower.GetChannelStart(); ch <= calibratedPower.GetChannelEnd(); ch++)
+        if ((str = strtok(nullptr, kCommaDelimiter)) != nullptr)
         {
-            SuccessOrExit(error = otPlatRadioAddCalibratedPower(gInstance, ch, calibratedPower.GetActualPower(),
-                                                                calibratedPower.GetRawPowerSetting().GetData(),
-                                                                calibratedPower.GetRawPowerSetting().GetLength()));
+            SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint32(str, channelMask));
+            aChannelMask = channelMask;
+            error        = OT_ERROR_NONE;
+            break;
         }
     }
 
+exit:
+    return error;
+}
+
+otError Config::UpdateChannelMasks(const Power::Domain &aDomain)
+{
+    otError error = OT_ERROR_NONE;
+
+    if (mProductConfigFile.HasKey(kKeySupportedChannelMask))
+    {
+        SuccessOrExit(error = GetChannelMask(kKeySupportedChannelMask, aDomain, mSupportedChannelMask));
+    }
+
+    if (mProductConfigFile.HasKey(kKeySupportedChannelMask))
```

Comment:
```suggestion
    if (mProductConfigFile.HasKey(kKeyPreferredChannelMask))
```
Can we add a test to cover this case?

Reply:
I have added `tests/scripts/expect/posix-channel-mask.exp` to test this case.